### PR TITLE
feat(runner-binaries-syncer): Makes Optional

### PR DESCRIPTION
**Pull request by mcaulifn**
- Makes module optional, defaults to true
- Default userdata does not download if syncer lambda not deployed
- `s3_bucket_runner_binaries` is changed to `s3_bucket_runner_binaries_arn`. This is a breaking change if just using the `runners` module.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -340,3 +340,9 @@ variable "runner_additional_security_group_ids" {
   type        = list(string)
   default     = []
 }
+
+variable "sync_runner_binary" {
```
Comment (npalm) [inline variables.tf:344]: I prefer a name like `runner-binaries-syncer`

File: modules/runners/variables.tf [hunk 2]
```diff
@@ -34,10 +34,15 @@ variable "environment" {
   type        = string
 }
 
-variable "s3_bucket_runner_binaries" {
-  type = object({
-    arn = string
-  })
+variable "s3_bucket_runner_binaries_arn" {
+  description = "Runner binary S3 bucket ARN."
+  #type = object({
```
Comment (npalm) [inline modules/runners/variables.tf:39]: Some commented code, I have to look in more details to variable changes

File: modules/runners/main.tf [hunk 3]
```diff
@@ -107,10 +108,13 @@ resource "aws_launch_template" "runner" {
 locals {
   arm_patch = var.runner_architecture == "arm64" ? templatefile(local.userdata_arm_patch, {}) : ""
   install_config_runner = templatefile(local.userdata_install_config_runner, {
-    environment                     = var.environment
-    s3_location_runner_distribution = var.s3_location_runner_binaries
-    run_as_root_user                = var.runner_as_root ? "root" : ""
-    arm_patch                       = local.arm_patch
+    environment      = var.environment
```
Comment (npalm) [inline modules/runners/main.tf:111]: Looks like formatting is broken, but guess this is github. Should be covered by CI

Comment (mcaulifn) [reply modules/runners/main.tf:111]: This was after a `terraform fmt`. There must be something about the indenting of items in a function call.

File: modules/runners/variables.tf (hunk 2 above)
Comment (mcaulifn) [reply modules/runners/variables.tf:39]: Will clean this up.

File: modules/runners/variables.tf [hunk 4]
```diff
@@ -322,3 +327,9 @@ variable "runner_additional_security_group_ids" {
   type        = list(string)
   default     = []
 }
+
+variable "sync_runner_binary" {
```
[the remark this answers is not in the document]
Comment (mcaulifn) [reply modules/runners/variables.tf:331]: I named it this way so as not to confuse with the module. Are you concerned with the overlap?

**Comment by ScottGuymer** [comment]
Is this still something that is wanted?

As part of #1444 I have added in something similar to the userdata script. To enable a download directly from GH in the prebuilt image.

https://github.com/philips-labs/terraform-aws-github-runner/pull/1444/files#diff-af0a08cb08ce2e8212cd4ec1f6e29d494e547eb476cf858e334570d77011facbR19

**Comment by npalm** [comment]
I would opt for closing this one.

**Comment by npalm** [comment]
@mcaulifn thanks for your contribution, but will close the PR for now. Feel free to re-open if you have any need.

**Pull request closed without merging**